basicdataset/src/test/java/ai/djl/basicdataset/GoEmotionsTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.basicdataset;
+
+import ai.djl.basicdataset.nlp.GoEmotions;
+import ai.djl.basicdataset.utils.TextData;
+import ai.djl.ndarray.NDManager;
+import ai.djl.repository.Repository;
+import ai.djl.training.dataset.Dataset;
+import ai.djl.training.dataset.Record;
+import ai.djl.translate.TranslateException;
+import java.io.IOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class GoEmotionsTest {
+
+    private static final int EMBEDDING_SIZE = 15;
+
+    @Test
+    public void testGoEmotionTrainLocal() throws IOException, TranslateException {
```

Comment:
I notice you have three tests that are almost the same. Instead, it might be better to put them through a loop instead of copying and pasting. The technically correct thing to do is use the testng parameterized tests, but it is fine if you just do a `for (Usage usage : {Usage.Train, Usage.Validation, Usage.Test})` loop. You can then put the expected sizes in a switch statement.